scripts/add_extra_components.py

```diff
@@ -222,6 +223,128 @@ def attach_stores(n, costs, config):
         )
 
 
+def attach_cooking_technologies(n, cooking_costs, config):
+    if config["clean_cooking"]["enable"] != True:
+        return
+
+    carriers = config["clean_cooking"]["fuel"]
+    cooking_costs = cooking_costs
+    buses_i = n.buses.index
+    buses_i = [
+        bus for bus in buses_i if not (bus.endswith("battery") or bus.endswith("H2"))
+    ]
+
+    bus_sub_dict = {k: n.buses[k].values for k in ["x", "y", "country"]}
+    for key in bus_sub_dict:
+        bus_sub_dict[key] = bus_sub_dict[key][: len(buses_i)]
+
+    cooking_buses_i = None
+    if "heat" in carriers:
+        cooking_buses_i = n.madd(
+            "Bus", [bus + " cooking" for bus in buses_i], carrier="heat", **bus_sub_dict
+        )
+
+    for fuel in carriers:
+        if fuel == "heat":
+            continue
+
+        if fuel == "AC" and "AC" in n.buses.carrier.values:
+            fuel_buses_i = n.buses.index[n.buses.carrier == "AC"]
```

Comment:
Ok when AC, what happens with `fuel_buses_i`?

Reply:
Here I am setting the index of the AC cooking bus to those of the buses for AC since you mentioned that we already have an AC carrier which implies the existence of AC buses as well.